ftw/copymovepatches/tests/test_move_updates_modified_date.py

```diff
@@ -9,50 +9,61 @@
 
 class TestMoveUpdatesModifiedDate(FunctionalTestCase):
 
-    def test_moving_AT_object_updates_modified_date(self):
+    date_name = 'modified'
+
+    def test_moving_AT_object_updates_date(self):
         self.grant('Manager')
         old_parent = create(Builder('folder').titled(u'Old Parent'))
         new_parent = create(Builder('folder').titled(u'New Parent'))
 
         with freeze(datetime(2000, 1, 1)) as clock:
             folder = create(Builder('folder').within(old_parent))
-            self.assert_modified_date(datetime(2000, 1, 1), folder)
+            self.assert_date(datetime(2000, 1, 1), folder)
 
             clock.forward(days=4)
             new_parent.manage_pasteObjects(old_parent.manage_cutObjects(
                 [folder.getId()]))
             folder = new_parent.get(folder.getId())
-            self.assert_modified_date(datetime(2000, 1, 5), folder)
+            self.assert_date(datetime(2000, 1, 5), folder)
 
-    def test_moving_DX_object_updates_modified_date(self):
+    def test_moving_DX_object_updates_date(self):
         self.grant('Manager')
         old_parent = create(Builder('dx container').titled(u'Old Parent'))
         new_parent = create(Builder('dx container').titled(u'New Parent'))
 
         with freeze(datetime(2000, 1, 1)) as clock:
             folder = create(Builder('dx container').within(old_parent))
-            self.assert_modified_date(datetime(2000, 1, 1), folder)
+            self.assert_date(datetime(2000, 1, 1), folder)
 
             clock.forward(days=4)
             new_parent.manage_pasteObjects(old_parent.manage_cutObjects(
                 [folder.getId()]))
             folder = new_parent.get(folder.getId())
-            self.assert_modified_date(datetime(2000, 1, 5), folder)
+            self.assert_date(datetime(2000, 1, 5), folder)
 
-    def assert_modified_date(self, expected_date, obj):
+    def assert_date(self, expected_date, obj):
         expected_date = DateTime(expected_date)
-        self.assertEquals(expected_date,
-                          DateTime(obj.modified()),
-                          'Wrong modification date on object.')
-        self.assert_modified_date_in_catalog(expected_date, obj)
+        self.assertEquals(
+            expected_date,
+            DateTime(getattr(obj, self.date_name)()).toZone(DateTime().timezone()),
```

Comment:
it might be a good idea to put retrieving the attribute from the object under test into its own method which can then be overridden by subclasses. In our current two cases you're lucky as both are methods. 

in `assert_date` you'd have:

```
self.assertEquals(
    expected_date,
    DateTime(self.retrieve_indexer_value(obj)).toZone(DateTime().timezone()),
```

and in the subclass that tests for a certain index

```
def retrieve_indexer_value(self, obj):
    return obj.modified()
```

Reply:
Did an in between your solution an mine 😸, so that I only have to overwrite the method if the date cannot be retrieved from a method call:
```
    def get_date_on_object(self, obj):
        return getattr(obj, self.date_name)()
```